core/services/relay/evm/cap_encoder.go

```diff
@@ -84,41 +84,65 @@ func (c *capEncoder) Encode(ctx context.Context, input values.Map) ([]byte, erro
 }
 
 func prependMetadataFields(meta consensustypes.Metadata, userPayload []byte) ([]byte, error) {
-	// TODO: use all 7 fields from Metadata struct
-	result := []byte{}
-	workflowID, err := decodeID(meta.WorkflowID, idLen)
-	if err != nil {
+	var err error
+	var result []byte
+
+	// 1. Version (1 byte)
+	if meta.Version > 255 {
+		return nil, fmt.Errorf("version must be between 0 and 255")
+	}
+	result = append(result, byte(meta.Version))
+
+	// 2. Execution ID (32 bytes)
+	if result, err = decodeAndAppend(meta.ExecutionID, 32, result, "ExecutionID"); err != nil {
 		return nil, err
 	}
-	result = append(result, workflowID...)
 
-	donID, err := decodeID(meta.DONID, 4)
-	if err != nil {
+	// 3. Timestamp (4 bytes)
+	tsBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(tsBytes, meta.Timestamp)
+	result = append(result, tsBytes...)
+
+	// 4. DON ID (4 bytes)
+	if result, err = decodeAndAppend(meta.DONID, 4, result, "DONID"); err != nil {
 		return nil, err
 	}
-	result = append(result, donID...)
 
-	executionID, err := decodeID(meta.ExecutionID, idLen)
-	if err != nil {
+	// 5. DON config version (4 bytes)
+	cfgVersionBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(cfgVersionBytes, meta.DONConfigVersion)
+	result = append(result, cfgVersionBytes...)
+
+	// 5. Workflow ID / spec hash (32 bytes)
+	if result, err = decodeAndAppend(meta.WorkflowID, 32, result, "WorkflowID"); err != nil {
 		return nil, err
 	}
-	result = append(result, executionID...)
 
-	workflowOwner, err := decodeID(meta.WorkflowOwner, 20)
-	if err != nil {
+	// 6. Workflow Name (10 bytes)
+	if result, err = decodeAndAppend(meta.WorkflowName, 10, result, "WorkflowName"); err != nil {
```

Comment:
the check i wrote this morning has the workflow name as plain text. 
https://github.com/smartcontractkit/chainlink/commit/2cc94eb1d0c1e4113865c89deeff7ae66ec15693
i'm not sure if meta.WorkflowName is hex or not by the time it gets here.

Reply:
that's stale commits from a previous PR, I'm not doing any changes to cap_encoder. I'll go away after a rebase